plugins/module_utils/redfish_utils.py

```diff
@@ -1601,19 +1601,31 @@ def set_bios_attributes(self, attributes):
 
         # Make a copy of the attributes dict
         attrs_to_patch = dict(attributes)
+        # List to hold attributes not found
+        attrs_bad = {}
 
         # Check the attributes
-        for attr in attributes:
-            if attr not in data[u'Attributes']:
-                return {'ret': False, 'msg': "BIOS attribute %s not found" % attr}
+        for attr_name, attr_value in attributes.items():
+            # Check if attribute exists
+            if attr_name not in data[u'Attributes']:
+                # Remove and proceed to next attribute if this isn't valid
+                attrs_bad.update({attr_name: attr_value})
+                del attrs_to_patch[attr_name]
+                continue
+
             # If already set to requested value, remove it from PATCH payload
-            if data[u'Attributes'][attr] == attributes[attr]:
-                del attrs_to_patch[attr]
+            if data[u'Attributes'][attr_name] == attributes[attr_name]:
+                del attrs_to_patch[attr_name]
+
+        warning = ""
+        if attrs_bad:
+            warning = "Incorrect attributes %s" % (attrs_bad)
 
         # Return success w/ changed=False if no attrs need to be changed
         if not attrs_to_patch:
             return {'ret': True, 'changed': False,
-                    'msg': "BIOS attributes already set"}
+                    'msg': "BIOS attributes already set",
+                    'warning': warning}
```

Comment:
Please call `module.warn()` instead of returning `warning`. (For that `module` needs to be passed here.)

Reply:
Thanks for pointing that convention to us; I agree we should be using `module.warn()`.

Reply:
Instead of changing the functions, I changed the module `exit_json`, removed the `warning` variable and added a separate `module.warn()`. Added changelog fragment as well. 

Warning message looks like
```
[WARNING]: Incorrect attributes {'TelemetryCUPS.1.EnableTelemetry': 'Enabled'}
[WARNING]: Incorrect attributes {'BootMode2': 'Uefi'}
```